tests/js/test_datatable.js

```diff
@@ -0,0 +1,59 @@
+
+QUnit.test('datatable', function(assert) {
+
+    assert.equal(
+        merge_on(
+            [
+                [{sample_id: 'sample_1', thing: 1}, {sample_id: 'sample_2', thing: 2}, {sample_id: 'sample_3', thing: 3}],
+                [{sample_id: 'sample_3', thang: 2}, {sample_id: 'sample_1', thang: 3}, {sample_id: 'sample_2', thang: 4}]
+            ],
+            'sample_id'
+        ).toString(),
+        [
+            {sample_id: 'sample_1', thing: 1, thang: 3},
+            {sample_id: 'sample_2', thing: 2, thang: 4},
+            {sample_id: 'sample_3', thing: 3, thang: 2}
+        ].toString()
+    );
```

Comment:
Could you add a test showing that the second array overwrites the first one when keys are the same?